File: backend-java/DZShop/cart-service/src/main/java/be/pxl/services/model/EnergyRating.java
package be.pxl.services.model;

public enum EnergyRating {
    A_PLUS_PLUS, // Best energy efficiency
    A_PLUS,
    A,
    B,
    C,  // Lower ratings
    D,
    E;

    // You can also add custom methods, like converting from a string or getting descriptions
    public static EnergyRating fromString(String value) {
        switch (value.toUpperCase()) {
            case "A++": return A_PLUS_PLUS;
            case "A+": return A_PLUS;
            case "A": return A;
            case "B": return B;
            case "C": return C;
            case "D": return D;
            case "E": return E;
            default: throw new IllegalArgumentException("Unknown rating: " + value);
        }
    }
}

File: frontend-web/dzshop-frontend/lib/energy-rating.ts
export enum EnergyRating {
  A_PLUS_PLUS = "A++",
  A_PLUS = "A+",
  A = "A",
  B = "B",
  C = "C",
  D = "D",
  E = "E",
}

export function fromString(value: string): string {
  switch (value.toUpperCase()) {
    case "A++":
      return "A_PLUS_PLUS";
    case "A+":
      return "A_PLUS";
    case "A":
      return "A";
    case "B":
      return "B";
    case "C":
      return "C";
    case "D":
      return "D";
    case "E":
      return "E";
    default:
      throw new Error(`Unknown rating: ${value}`);
  }
}

export function toString(value: string): string {
  switch (value) {
    case "A_PLUS_PLUS":
      return "A++";
    case "A_PLUS":
      return "A+";
    case "A":
      return "A";
    case "B":
      return "B";
    case "C":
      return "C";
    case "D":
      return "D";
    case "E":
      return "E";
    default:
      throw new Error(`Unknown rating: ${value}`);
  }
}

export const getAllEnergyRatings = (): { value: string; label: string }[] => {
  return Object.values(EnergyRating).map((rating) => ({
    value: fromString(rating),
    label: rating,
  }));
};
